build/Targets/Settings.targets

```diff
@@ -128,17 +128,31 @@
       </PropertyGroup>
     </Otherwise>
   </Choose>
+  
+  <!-- This is a really hacky way to detect whether we are on a legacy or a willow based VS install.
+       Basically, we check for a registry key that will only exist in legacy VS installs, and assume
+       we are a willow based installation if our VSVersion is 15.0 and the registry key doesn't exist. -->
```

Comment:
Another alternative to detect: VsDevCmd.bat in Willow install sets the environment variable "VSCMD_VER", the full VS version. This isn't set with Dev15 preview3 command prompt.